### PR TITLE
feat: add permission checking logic to the notification group service

### PR DESCRIPTION
**Pull request by CeerDecy**
#### What this PR does / why we need it:

Add permission checking logic to the notification group service

#### Specified Reviewers:

/assign @iutx 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |      Add permission checking logic to the notification group service        |
| 🇨🇳 中文    |       通知组服务添加权限校验逻辑       |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: internal/core/messenger/notifygroup/group.service.go [hunk 1]
```diff
@@ -391,3 +401,20 @@ func (n *notifyGroupService) checkNotifyPermission(userId, scopeType, scopeId, a
 	}
 	return nil
 }
+
+func (n *notifyGroupService) getIdentityInfo(ctx context.Context) (apistructs.IdentityInfo, error) {
```
Comment (iutx) [inline internal/core/messenger/notifygroup/group.service.go:405]: Please verify whether there are reusable elements in the project.

File: internal/apps/msp/apm/notifygroup/provider.go [hunk 2]
```diff
@@ -58,8 +60,17 @@ func (p *provider) Init(ctx servicehub.Context) error {
 	p.mspTenantDB = &db.MSPTenantDB{DB: p.DB}
 	p.monitorDB = &db2.MonitorDb{DB: p.DB}
 	if p.Register != nil {
-		type NotifyService = pb.NotifyGroupServiceServer
+		type NotifyService = notifygroup.NotifyGroupServiceServer
```
Comment (iutx) [inline internal/apps/msp/apm/notifygroup/provider.go:63]: Err protobuf source

File: internal/core/messenger/notifygroup/group.service.go [hunk 3]
```diff
@@ -193,7 +199,8 @@ func (n *notifyGroupService) GetNotifyGroup(ctx context.Context, request *pb.Get
 		return nil, errors.NewInternalServerError(err)
 	}
 	userIdStr := apis.GetUserID(ctx)
-	err = n.checkNotifyPermission(userIdStr, notifyGroup.ScopeType, notifyGroup.ScopeID, apistructs.GetAction)
+	ctx = transport.WithHeader(ctx, metadata.New(map[string]string{httputil.InternalHeader: "true"}))
```
Comment (iutx) [inline internal/core/messenger/notifygroup/group.service.go:202]: Incorrect internal-client injection

**Review by iutx — COMMENTED** [review]
@CeerDecy

File: internal/core/messenger/notifygroup/group.service.go (hunk 3 above)
Comment (CeerDecy) [reply internal/core/messenger/notifygroup/group.service.go]: all done

**Review by iutx — APPROVED** [review]
LGTM @sfwn  
The code change depends on the frontend pull request.

**Comment by iutx** [comment]
Please ensure that the CI checks pass. @CeerDecy

**Review by iutx — APPROVED** [review]
verified. @sfwn

File: internal/apps/msp/apm/notifygroup/notifygroup.service.go [hunk 4]
```diff
@@ -82,6 +85,7 @@ func (n *notifyGroupService) CreateNotifyGroup(ctx context.Context, request *pb.
 		return nil, errors.NewInternalServerError(err)
 	}
 	c := utils.NewContextWithHeader(ctx)
+	c = transport.WithHeader(c, metadata.New(map[string]string{httputil.InternalHeader: "true"}))
```
[the remark this answers is not in the document]
Comment (CeerDecy) [reply internal/apps/msp/apm/notifygroup/notifygroup.service.go]: done

File: internal/apps/msp/apm/notifygroup/notifygroup.service.go [hunk 5]
```diff
@@ -84,8 +82,7 @@ func (n *notifyGroupService) CreateNotifyGroup(ctx context.Context, request *pb.
 	if err != nil {
 		return nil, errors.NewInternalServerError(err)
 	}
-	c := utils.NewContextWithHeader(ctx)
-	c = transport.WithHeader(c, metadata.New(map[string]string{httputil.InternalHeader: "true"}))
+	c := apis.WithInternalClientContext(ctx, discover.MSP())
```
Comment (sfwn) [inline internal/apps/msp/apm/notifygroup/notifygroup.service.go:85]: `discover.MSP()` will return the address of msp.
You should use `discover.SvcMSP` instead.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**